sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -29,19 +30,32 @@ namespace SonarAnalyzer.Helpers
     public class SonarAnalysisContext
     {
         private readonly AnalysisContext context;
-        public static Func<SyntaxTree, bool> ShouldAnalysisBeDisabled { get; set; }
+        private readonly IEnumerable<DiagnosticDescriptor> supportedDiagnostics;
+
+        /// <summary>
+        /// This delegate is only set by the VSIX when the projects have the NuGet package installed to avoid repeated behaviors.

```

Comment:
This comment isn't accurate any more, now that we've using the same mechanism to disable specific rules.